### ISSUE TITLE
C# formatter: Add a setting for space between empty braces

### ISSUE THREAD

**Issue by aaronfranke** [dotnet/roslyn#55061]
**Brief description:**

The C# code formatter in VS Code (which I'm pretty sure is this repository) has some settings in `.editorconfig` that allow configuring various settings for the formatter. We currently have `csharp_space_between_empty_square_brackets`. We should also add a similar setting for braces, perhaps `csharp_space_between_empty_braces`.

**Languages applicable:**

C#

**Code example that the analyzer should report:**

Currently, the formatter in VS Code automatically does this change:

```diff
-    public class ToolAttribute : Attribute {}
+    public class ToolAttribute : Attribute { }
```

I would like to be able to specify a setting like `csharp_space_between_empty_braces = false` so that the formatter won't do this.

**Additional information:**

See this existing documentation for the current setting for brackets: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/formatting-rules#spacing-options

On a related note, there's also `csharp_space_within_single_line_array_initializer_braces` which is supported by ReSharper and Rider but not by VS or VS Code.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by sharwell** [comment]
This will likely fall under #31691

**Comment by CyrusNajmabadi** [comment]
tracking through https://github.com/dotnet/roslyn/issues/31691
